spec.html

```diff
@@ -4488,53 +4488,71 @@ <h2>Syntax</h2>
+            1. Return *+&infin;*<sub>𝔽</sub>.
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits `.` DecimalDigits? ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of the first |DecimalDigits|.
+            1. If the second |DecimalDigits| is present, let _b_ be MV of the second |DecimalDigits|. Otherwise, let _b_ be 0.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Let _n_ be the number of code points in the second |DecimalDigits|.
+            1. Return RoundStringMVResult((_a_ + (_b_ &times; 10<sup>-_n_</sup>)) &times; 10<sup>_e_</sup>).
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: `.` DecimalDigits ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of |DecimalDigits|.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Let _n_ be the number of code points in |DecimalDigits|.
+            1. Return RoundStringMVResult(_a_ &times; 10<sup>_e_-_n_</sup>).
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of |DecimalDigits|.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Return RoundStringMVResult(_a_ &times; 10<sup>_e_</sup>).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-roundstringmvresult" aoid="RoundStringMVResult">
+          <h1>RoundStringMVResult ( _n_ )</h1>
+          <p>The abstract operation RoundStringMVResult takes argument _n_ (a mathematical value). It converts _n_ to a Number in an implementation-defined manner. For the purposes of this abstract operation, a digit is significant if it is not zero or there is a non-zero digit to its left and there is a non-zero digit to its right. For the purposes of this abstract operation, "the mathematical value denoted by" a representation of a mathematical value is the inverse of "the decimal representation of" a mathematical value. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If the decimal representation of _n_ has fewer than 20 significant digits, return 𝔽(_n_).
+            1. Let _option1_ be the mathematical value denoted by the result of replacing each significant digit in the decimal representation of _n_ after the 20th with a 0 digit.
+            1. Let _option2_ be the mathematical value denoted by the result of replacing each significant digit in the decimal representation of _n_ after the 20th with a 0 digit and then incrementing it at the 20th position (with carrying as necessary).
+            1. Let _chosen_ be an implementation-defined choice of either _option1_ or _option2_.
+            1. Return 𝔽(_chosen_).
+          </emu-alg>
```

Comment:
Before the call to RoundStringMVResult, we convert a decimal representation to a mathematical value, and then inside the call, we convert that to a decimal representation, tweak that, then convert that to a mathematical value, and convert *that* to a Number. So it's:
- decimal rep -> math value -> decimal rep -> math value -> Number

I'm not saying it's wrong, it just seems circuitous.

Reply:
The first input (the input to MV) is a parse node, not a (typical) decimal representation of a mathematical value. I think this is fine, but I'd be open to alternative suggestions.